core/types/uuid.go

```diff
@@ -42,22 +43,48 @@ func (u UUID) String() string {
 	return uuid.UUID(u).String()
 }
 
-func (u UUID) Value() (driver.Value, error) {
-	return u[:], nil // []byte works for sql
+func (u *UUID) Value() (driver.Value, error) {
+	return u.String(), nil
 }
 
 func (u UUID) Bytes() []byte {
 	return u[:]
 }
 
-var _ driver.Valuer = UUID{}
+// Over json, we want to send uuids as strings
+func (u UUID) MarshalJSON() ([]byte, error) {
+	return json.Marshal(u.String())
+}
+
+func (u UUID) UnmarshalJSON(b []byte) error {
```

Comment:
Unmarshal/Scan type methods should almost always be pointer receiver on the other hand.

Reply:
```go

func Test_UUIDJSON(t *testing.T) {
	seed := []byte("test")

	uuid := types.NewUUIDV5(seed)

	b, err := json.Marshal(uuid)
	if err != nil {
		t.Fatal(err)
	}

	var uuid2 types.UUID
	err = json.Unmarshal(b, uuid2)
	if err != nil {
		t.Fatal(err)
	}
}
```

```
=== RUN   Test_UUIDJSON
    /home/jon/kwil/git/kwil-db/core/types/uuid_test.go:33: json: Unmarshal(non-pointer types.UUID)
--- FAIL: Test_UUIDJSON (0.00s)
FAIL
coverage: 0.8% of statements
FAIL	github.com/kwilteam/kwil-db/core/types	0.012s
```

Reply:
```go

func Test_UUIDJSON(t *testing.T) {
	seed := []byte("test")

	uuid := types.NewUUIDV5(seed)

	b, err := json.Marshal(uuid)
	if err != nil {
		t.Fatal(err)
	}

	t.Log(uuid)

	var uuid2 types.UUID
	err = json.Unmarshal(b, uuid2) // call of Unmarshal passes non-pointer as second argumentunmarshaldefault
	if err != nil {
		t.Error(err)
	}

	var uuid3 types.UUID
	err = json.Unmarshal(b, &uuid3)
	if err != nil {
		t.Fatal(err)
	}
	t.Log(uuid3) // 00000000-0000-0000-0000-000000000000
}
```

```
=== RUN   Test_UUIDJSON
    /home/jon/kwil/git/kwil-db/core/types/uuid_test.go:30: 24aa70cf-0e18-57c9-b449-da8c9db37821
    /home/jon/kwil/git/kwil-db/core/types/uuid_test.go:35: json: Unmarshal(non-pointer types.UUID)
    /home/jon/kwil/git/kwil-db/core/types/uuid_test.go:43: 00000000-0000-0000-0000-000000000000
--- FAIL: Test_UUIDJSON (0.00s)
FAIL
coverage: 2.8% of statements
FAIL	github.com/kwilteam/kwil-db/core/types	0.013s
```